src/Compilers/CSharp/Test/Semantic/Semantics/RecordTests.cs

```diff
@@ -18548,6 +18548,25 @@ record A
                 );
         }
 
+        [Fact]
+        [WorkItem(48723, "https://github.com/dotnet/roslyn/issues/48723")]
+        public void EqualityContract_24_SetterOnlyProperty()
+        {
+            var src = @"
+record R
+{
+    protected virtual System.Type EqualityContract { set { } }
```

Comment:
>set { } [](start = 53, length = 7)

Consider adding a test with a property that has both accessors, unless we already have one. I assume that should be a success scenario? Or should we disallow property like that?

Reply:
Also, consider adding test(s) when a record class has only setter or both accessors in metadata and we try to derive another record from it with explicit/implicit definition of EqualityContract. 

---
In reply to: [508933817](https://github.com/dotnet/roslyn/pull/48798#discussion_r508933817) [](ancestors = 508933817)

Reply:
> Or should we disallow property like that [with both accessors]?

I think it's okay to allow, as it meets our requirements and doesn't get in the way of what we need to do. 

---
In reply to: [508941602](https://github.com/dotnet/roslyn/pull/48798#discussion_r508941602) [](ancestors = 508941602,508933817)